extensions/wikia/EditAccount/SpecialEditAccount_body.php

```diff
@@ -518,4 +529,55 @@ public static function generateRandomScrambledPassword() {
 		$REQUIRED_CHARS = "A1a";
 		return (wfGenerateToken() . $REQUIRED_CHARS);
 	}
+
+	/** Hook for storing historical log of email changes **/
+	public static function logEmailChanges($user, $new_email, $old_email) {
+		global $wgExternalSharedDB, $wgUser, $wgRequest;
+		if ( $wgExternalSharedDB ) {
+			$dbw = wfGetDB( DB_MASTER, array(), $wgExternalSharedDB );
+			$dbw->insert(
+				'user_email_log',
+				['user_id' => $user->getId(),
+				 'old_email' => $old_email,
+				 'new_email' => $new_email,
+				 'changed_by_id' => $wgUser->getId(),
+				 'changed_by_ip' => $wgRequest->getIP()  // stored as string
+				]);
+		}
+		return true;
+	}
+
+	public function displayLogData() {
+		global $wgExternalSharedDB, $wgOut, $wgRequest;
+
+		$user_id = $wgRequest->getVal('user_id', null);
+		$rows = [];
+
+		if ( $wgExternalSharedDB && $user_id ) {
+			$dbr = wfGetDB ( DB_SLAVE, array(), $wgExternalSharedDB );
+			$res = $dbr->select (
+				'user_email_log',   // from
+				["*"], 				// cols
+				"user_id = " . $dbr->strencode($user_id), // where
+				__METHOD__,
+				["ORDER BY" => "changed_at DESC"]  // options
+				);
+			while ( $row = $dbr->fetchObject( $res ) ) {
+				$row->user_name = User::newFromID($row->user_id);
```

Comment:
Since this shouldn't change as we're querying on the user ID, can this just be set before the query as a single variable that can then also be used to pass the user name to the template?